File: src/main/java/no/nav/pto/veilarbfilter/repository/FilterService.java
package no.nav.pto.veilarbfilter.repository;

import no.nav.pto.veilarbfilter.domene.FilterModel;
import no.nav.pto.veilarbfilter.domene.NyttFilterModel;
import no.nav.pto.veilarbfilter.domene.SortOrder;

import java.util.List;
import java.util.Optional;

public interface FilterService {
    Optional<FilterModel> lagreFilter(String filterBrukerId, NyttFilterModel nyttFilter);

    Optional<FilterModel> oppdaterFilter(String filterBrukerId, FilterModel filter);

    Optional<FilterModel> hentFilter(Integer filterId);

    List<FilterModel> finnFilterForFilterBruker(String filterBrukerId);

    Integer slettFilter(Integer filterId, String filterBrukerId);

    List<FilterModel> lagreSortering(String filterBrukerId, List<SortOrder> sortOrder);
}

File: src/main/no/nav/pto/veilarbfilter/service/FilterService.kt
package no.nav.pto.veilarbfilter.service


import no.nav.pto.veilarbfilter.model.FilterModel
import no.nav.pto.veilarbfilter.model.NyttFilterModel
import no.nav.pto.veilarbfilter.model.SortOrder

interface FilterService {
    suspend fun lagreFilter(filterBrukerId: String, nyttFilter: NyttFilterModel): FilterModel?
    suspend fun oppdaterFilter(filterBrukerId: String, filter: FilterModel): FilterModel
    suspend fun hentFilter(filterId: Int): FilterModel?
    suspend fun finnFilterForFilterBruker(filterBrukerId: String): List<FilterModel>
    suspend fun slettFilter(filterId: Int, filterBrukerId: String): Int
    suspend fun lagreSortering(filterBrukerId: String, sortOrder: List<SortOrder>): List<FilterModel>
}
